### ISSUE TITLE
Overriding path to support multiple methods on the same /resource

### ISSUE THREAD

**Issue by jbmusso** [Urigo/SOFA#51]
Thanks for creating this library, very nice idea!

I'm trying to support both `GET` and `HEAD` for the same resource, for example `"foobar"`, mounted on `/foobar` path. The v0.4.0 release with HTTP method customization comes in handy.

I'm trying to expose the following HTTP routes:
* use `HEAD /foobar`, semantically, to check for existence/absence of a given resource, and return respectively code `200` and `404` (empty body)
* use `GET /foobar` to fully retrieve the content of that resource, returning a `200` + body in case of existence, and a `404` if missing

I use two GraphQL queries defined as `foobar()` and `foobarExists()`, respectively, in my GraphQL schema. I need to override the path mounted in express as `/foobar` instead of `/foobar-exists`.

At the moment, I simply strip the `-exists` part of the path (`@directive()` would be cleaner, rather than using a convention in the `path`/GraphQL query name). However this approach with current version of SOFA only seems to update the path in the swagger document, but does not override the path in the `express.Router` instance. By looking at SOFA's code, my understanding is that this cannot be supported at the moment unless manually updating the `express.Router` instance returned by `useSofa()`.

In short, I'm trying to expose multiple GraphQL queries on the same `/resource` path, semantically distinguishing actions via HTTP methods.
What would you recommend for such use case?

**Comment by Urigo** [comment]
Adding the option to do that doesn't seem like a big change,
maybe you can give it a try with a PR?
also show an example of how you imagine the API to look

thanks

**Comment by Pruxis** [comment]
@Urigo I have the same use case for work, and was wondering what the desired approach would be. Configuration for queries / mutations their path in the sofa middleware or utilising directives?

**Comment by Urigo** [comment]
@Pruxis it would be awesome to get a contribution!

I feel like passing a config is the better option here but if you think differently I'm open for other implementations

**Comment by LRipi** [comment]
Any update about this feature ?

**Comment by Urigo** [comment]
@LRipi we are still looking for a production use case we could test it out with and maybe even a contributor.
Are you able to provide one of those?

**Comment by LRipi** [comment]
@Urigo I can make a basic repository with a use case! 
I will make a repository and send you the link when it will be finished.

**Comment by Urigo** [comment]
Thank you @LRipi !

**Comment by LRipi** [comment]
@Urigo I made a quick example here -> https://github.com/LRipi/sofa-graphql

## Quick things to know 

- This is an integration with GraphQL Nexus and Sofa (many thing could be scaled and optimize)
- The README.md details the content of the repository

## Things that could be awesome to have

- The other HTTP method (e.g. PUT, PATCH, DELETE)
- The route to use (see graphql/Post/Mutation and graphql/Post/Query for more info in comments)
- The parameters to set as path parameters (instead of body parameters)

## Where are the improvements?

In the Posts' query, you'll find an exemple of query with graphQL nexus [here](https://github.com/LRipi/sofa-graphql/blob/master/api/graphql/Post/Query.ts), I added few comments to let you know which route we can imagine to create in REST.
Same thing in the Posts' mutation, example [here](https://github.com/LRipi/sofa-graphql/blob/master/api/graphql/Post/Mutation.ts).

In the other side, where we implement SOFA, you'll find some comments on the initialization section ([this part](https://github.com/LRipi/sofa-graphql/blob/master/api/rest/index.ts#L50)).

Hit me up if you have some questions or if you want to discuss about the code or ideas for this feature!

**Comment by LRipi** [comment]
@Urigo Any news about this feature?
